app/models/call_session.rb

```diff
@@ -1,33 +1,53 @@
 class CallSession
-  KEY = 'CALL-SESSION'
   EXPIRE_CALL_SESSION_IN = 24.hours
+  COUNTRY = 'IN'
 
-  def initialize(user_phone_number, patient_phone_number)
-    @user = User.find_by_phone_number(user_phone_number)
+  attr_reader :patient_phone_number, :user
+
+  def initialize(call_id, user_phone_number, patient_phone_number)
+    @call_id = call_id
+    @user = User.find_by_phone_number(sanitized_phone_number(user_phone_number))
     @patient_phone_number = PatientPhoneNumber.find_by_number(patient_phone_number)
   end
 
-  def save(call_id)
-    Rails.cache.write(session_key(call_id),
+  def authorized?
+    @user.present? &&
+      patient_phone_number.present? &&
+      @user.phone_number != patient_phone_number.number
+  end
+
+  def save
+    Rails.cache.write(CallSession.session_key(@call_id),
```

Comment:
We should keep this outside the rails cache. A better option would be to use a redis store.

Reply:
Yeah, we want to do this too. We'll keep this separate from `Rails.cache` when we run the Redis PR.

Reply:
We will move to Redis in the future. Will need changes to infra, terraform, etc. before we can do that.